storage/csi-snapshot.md

```diff
@@ -0,0 +1,120 @@
+
+We will be releasing the `external-snapshotter` sidecar image to enable CSI Snapshot support in OCP as Developer Preview only. This sidecar can be used by internal teams, such as CNV or OCS, for their drivers.
+
+### User Stories [optional]
+
+Similar to how users and administrators provision PersistentVolume and PersistentVolumeClaim resources, VolumeSnapshotContent and VolumeSnapshot API resources will be included to create volume snapshots for users and administrators. A VolumeSnapshotContent is a snapshot taken from a volume in the cluster that has been provisioned by an administrator. It is a resource in the cluster, similar to a PersistentVolume. A VolumeSnapshot is a request for snapshot of a volume by a user. It is similar to a PersistentVolumeClaim.
+
+In addition to the above, VolumeSnapshotClass resources can be used to describe the “class” of storage when provisioning a VolumeSnapshot.
+
+#### Story 1
+
+As an OCS developer, I want to release the `external-snapshotter` sidecar image so that CSI drivers can utilize volume snapshots.
+
+#### Story 2
+
+### Implementation Details/Notes/Constraints [optional]
+
+This feature has already been implemented upstream. Therefore, this feature requires a Kubernetes rebase of 1.15 or later.
+
+### Risks and Mitigations
+
+* Kubernetes 1.15+ rebase
+
+## Design Details
+
+### Test Plan
+
+We’ll want the e2e tests to run using CSI sidecars shipped as part of OCP. There is ongoing progress in the following links regarding testing OCP CSI sidecars:
+
+* https://github.com/openshift/origin/pull/23560
+* https://jira.coreos.com/browse/STOR-223
+Tests for this sidecar are currently defined upstream at https://github.com/kubernetes-csi/external-snapshotter/tree/master/pkg/controller .
+
+### Graduation Criteria
+
+##### Dev Preview -> Tech Preview
+
+Feature complete, including:
+Create/delete volume snapshots
+Create new volumes from a snapshot
+SnapshotContent Deletion/Retain Policy
+Snapshot Object in Use Protection
+Separate the common controller from the sidecar controller
+Add secrets field to list-snapshots RPC in the CSI spec. Add “data-source-secret” in create-volume intended for accessing the data source. Implement them in external-snapshotter and external-provisioner.
+Add metrics support
+Unit and e2e tests implemented
+Update snapshot CRDs to v1beta1 and enable VolumeSnapshotDataSource feature gate by default.
+
+##### Tech Preview -> GA 
+
+Snapshot feature is used as a basic building block in other advanced applications.
+Feature deployed in production and have gone through at least one K8s upgrade.
+
+##### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+### Version Skew Strategy
+
+## Implementation History
+
+## Drawbacks
+
+## Alternatives
+
+## Infrastructure Needed [optional]
+
+This requies a github repo be created under the openshift org to hold the CSI Snapshot code. The name of this repo is `csi-external-snapshot`.
+
```

Comment:
This repo already exists at https://github.com/openshift/csi-external-snapshotter

Reply:
I've updated this line to reference the existing repository.